src/contexts/MatrixClientContext.tsx

```diff
@@ -42,10 +34,9 @@ export function useMatrixClientContext(): MatrixClient {
 
 const matrixHOC = <ComposedComponentProps extends {}>(
     ComposedComponent: ComponentClass<ComposedComponentProps>,
-): ForwardRefExoticComponent<
-    PropsWithoutRef<Omit<ComposedComponentProps, "mxClient">> &
-        RefAttributes<InstanceType<ComponentClass<ComposedComponentProps>>>
-> => {
+): ((
+    props: Omit<ComposedComponentProps, "mxClient"> & React.RefAttributes<InstanceType<typeof ComposedComponent>>,
+) => React.ReactElement | null) => {
```

Comment:
this is some next-level cursed typescript. Help me grok what's going on here?

Reply:
It is matching the return type from `react.d.ts` workaround - more details at https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref/58473012#58473012